### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by erikjohnston**
We have tests so we may as well run them

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+on: [push, pull_request]
+
+name: Continuous integration
+
+jobs:
+  check:
+    name: Check
```
Comment (anoadragon453) [inline .github/workflows/tests.yaml:7]:
```suggestion
    name: Cargo Check
```

Just for some clarity, as "check" is a fairly generic word.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+on: [push, pull_request]
```
Comment (anoadragon453) [inline .github/workflows/tests.yaml:1]: I'm not sure if `pull_request` is necessary here if you already have push. I at least noticed on this PR that we seem to be running 8 checks in total, which is overkill.

Comment (erikjohnston) [reply .github/workflows/tests.yaml:1]: So I think `pull_request` actually merges in the base branch too, so they're not exactly equivalent. We can probably drop `push` and/or only do it for the `main` branch :shrug:

Comment (callahad) [reply .github/workflows/tests.yaml:1]: The common pattern is to run tests on pull requests and on pushes to your main / release branches. E.g., Django: https://github.com/django/django/blob/a24fed399ced6be2e9dce4cf28db00c3ee21a21c/.github/workflows/tests.yml#L3-L7

Comment (erikjohnston) [reply .github/workflows/tests.yaml:1]: Done!

Comment (callahad) [reply .github/workflows/tests.yaml:1]: ...I still see `on: [push, pull_request]`?

Comment (erikjohnston) [reply .github/workflows/tests.yaml]: ... my push failed :facepalm:

**Review by callahad — APPROVED** [review]
lgtm

**Pull request merged**